### PR TITLE
FINOS Security Scanning Tooling

### PR DESCRIPTION
**Pull request by TheJuanAndOnly99**
This PR enables the new [FINOS Security Scanning Tooling](https://github.com/finos/security-scanning) by adding 2 GitHub Actions:

1. [CVE scanning](https://github.com/TheJuanAndOnly99/waltz/blob/security-scanning/.github/workflows/cve-scanning.yml)
2. [Static Code Analysis](https://github.com/TheJuanAndOnly99/waltz/blob/security-scanning/.github/workflows/semgrep.yml)

As you can see in the PR I have addressed some CVEs however the [CVE Scanning Action fails](https://github.com/TheJuanAndOnly99/waltz/actions/runs/3290570369/jobs/5423588362) with the following error:

```
Error: jetty-server-9.4.18.v20190429.jar: CVE-2021-28165(7.5), CVE-2022-2048(7.5), CVE-2020-27216(7.0)
5189Error: websocket-common-9.4.18.v20190429.jar: CVE-2021-28165(7.5), CVE-2022-2048(7.5), CVE-2020-27216(7.0)
5190Error: websocket-server-9.4.18.v20190429.jar: CVE-2021-28165(7.5), CVE-2022-2048(7.5), CVE-2020-27216(7.0)
```
 
These errors seem to be related to the spark-core artifact. They can be fixed by updating spark.version from 2.9.1 to 2.9.4 however I did not go ahead with this because I saw that [there is a comment in the code](https://github.com/finos/waltz/blob/master/pom.xml#L101) stating that version 2.9.3 causes issues with tomcat. I would highly recommend that you try the new 2.9.4 version (updated just a few months ago) as the 2.9.1 version has some potentially serious vulnerabilities.

The Static Code Analysis Action has found [46 issues](https://github.com/TheJuanAndOnly99/waltz/actions/runs/3290406350/jobs/5423202373). Please take a look at these as not all may be relevant but we believe that enabling this tool can help to keep the Waltz codebases cleaner and more secure.

### CODE REVIEW

**Comment by davidwatkins73** [comment]
Hi @TheJuanAndOnly99 
Thanks for the PR., it looks great. 
A couple of questions/thoughts.... 

- not 100% sure relying on RC builds of libs is a good idea (jackson). 
- should we look to decom our usage of dependabot and snyk in favour of the FINOS suite?
- we've recently been upgrading to h2 v2 internally and have noticed some odd mirroring going on in on-site artifactory.  Hasn't been showing up in the standard mirror repo for some reason.  This is making the upgrade more painful as people have to use set up additional mirror entries in their `settings.xml`.  Not a biggie for us but would be curious if this has been observed elsewhere or is a db peculiarity.
- the spark dependency is a bit of a pain.  It only manifests itself when packaging in a war and deploying on Tomcat.  Looks like it's been reported a bunch of times and still no fix.  We love the DSL based approach of Spark, but may need to look to something like Javelin to remedy (initial thoughts look like it should be an easy, if laborious, switch over)

Thanks again for the PR.  We'll be looking to merge it early next week as we just want to get a few final fixes in place with the current code base before releasing 1.45

File: pom.xml [hunk 1]
```diff
@@ -82,8 +82,8 @@
         <freemarker.version>2.3.31</freemarker.version>
         <hikari.version>3.4.5</hikari.version>
         <immutables.version>2.9.2</immutables.version>
-        <jackson.version>2.13.3</jackson.version>
-        <jackson-jsr310.version>2.13.3</jackson-jsr310.version>
+        <jackson.version>2.14.0-rc2</jackson.version>
+        <jackson-jsr310.version>2.14.0-rc2</jackson-jsr310.version>
```
Comment (davidwatkins73) [inline pom.xml:86]: not 100% sure about listing RC builds.  It's fine for dev, for prod I'm not so sure.

**Comment by davidwatkins73** [comment]
Another comment.  It looks like the linked build broke because of the cve scanning.  Is there a way to disable override this in case we need to get a build out which has no viable fix.  Is that what the `allow_list.xml` file is used for ?  In which case is this the [correct docs](https://jeremylong.github.io/DependencyCheck/general/suppression.html) for it.  We'd need to see how we can get an exclusion on Spark in the short term.

**Comment by maoo** [comment]
> Hi @TheJuanAndOnly99 Thanks for the PR., it looks great. A couple of questions/thoughts....
> 
> * not 100% sure relying on RC builds of libs is a good idea (jackson).

Yes, agreed. We were just looking for a version that would solve the issue. I guess we could add the CVE to the allow list and update to 2.14.x version as soon as it's stable.

> * should we look to decom our usage of dependabot and snyk in favour of the FINOS suite?

It's not necessary in the short term, whatever works well for you; I may notice though, that if Snyk didn't spot the spark CVEs, then there's something wrong there.

Re. Dependabot, same story, I see many updates on Waltz, and I'm not sure why/how Dependabot missed it. I personally prefer Renovate (more about this on https://github.com/finos/security-scanning#dependency-update-tool

Eventually it's your choice which tool to use, as long as we can ensure that security scans happen in a reactive and proactive way.

> * we've recently been upgrading to h2 v2 internally and have noticed some odd mirroring going on in on-site artifactory.  Hasn't been showing up in the standard mirror repo for some reason.  This is making the upgrade more painful as people have to use set up additional mirror entries in their `settings.xml`.  Not a biggie for us but would be curious if this has been observed elsewhere or is a db peculiarity.

I'm not aware of it; is this something related with security scanning?

> * the Spark dependency is a bit of a pain.  It only manifests itself when packaging in a war and deploying on Tomcat.  Looks like it's been reported a bunch of times and still no fix (we checked the latest).  We love the DSL based approach of Spark, but may need to look to something like Javelin to remedy (initial thoughts look like it should be an easy, if laborious, switch over)

Agreed with the approach; the choice of libraries should involve also understanding if/how security is enforced, and which transitive dependencies are used.

> 
> Thanks again for the PR. We'll be looking to merge it early next week as we just want to get a few final fixes in place with the current code base before releasing 1.45

Sounds great.

> Another comment. It looks like the linked build broke because of the cve scanning. Is there a way to disable override this in case we need to get a build out which has no viable fix. Is that what the allow_list.xml file is used for ? In which case is this the [correct docs](https://jeremylong.github.io/DependencyCheck/general/suppression.html) for it. We'd need to see how we can get an exclusion on Spark in the short term.

Yes, some of our `pom.xml` updates should be reverted, and temporarily added to the `allow-list.xml`, to ensure that the build keeps working.

**Comment by davidwatkins73** [comment]
Just merged it.  Will start taking a look through the output.  Thanks for the PR.

**Comment by maoo** [comment]
Thanks @davidwatkins73 ! Would it be useful if I enable https://renovatebot.com/ for you on the Waltz repo? We're using it on several projects now and it works quite well! It saves a lot of time looking for the latest dependency versions, and creates Pull Requests for you.

**Pull request merged**